tokio-executor/src/global.rs

```diff
@@ -205,16 +215,26 @@ where
         let executor = unsafe { hide_lt(executor as &mut _ as *mut _) };
 
         cell.set(State::Ready(executor));
+    });
 
-        f(enter)
-    })
+    DefaultGuard {
```

Comment:
I would feel slightly more comfortable moving the creation of this up to where `Reset` was previously created. It shouldn't technically matter since none of the intervening code can panic (I don't think), but it does ensure that even if that code later changes, the thread-local will always be reset correctly.